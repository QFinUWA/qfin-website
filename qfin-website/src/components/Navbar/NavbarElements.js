import { FaBars } from 'react-icons/fa';
import { NavLink as Link } from 'react-router-dom';
import styled from 'styled-components';
  
// Style of the navbar
export const Nav = styled.nav`
  background: #24CC54;
  border-radius: 3px;
  height: 70px;
  display: flex;
  justify-content: center;
  padding: 0.2rem calc((100vw - 1000px) / 2);
  z-index: 12;
  /* Third Nav */
  /* justify-content: flex-start; */
`;

// Colour of text on nav bar
export const NavLink = styled(Link)`
  color: #313131;
  font-weight: bold;
  display: flex;
  align-items: center;
  text-decoration: none;
  padding: 0 1rem;
  height: 100%;
  cursor: pointer;
  &.active {
    color: #000000;
  }
`;

export const NavQFin = styled(Link)`
  color: #313131;
  font-weight: bold;
  display: flex;
  align-items: center;
  text-decoration: none;
  padding: 0 1rem;
  height: 100%;
  cursor: pointer;
  &.active {
    color: #000000;
  }
`;

// Icon for the hamburger menu
export const Bars = styled(FaBars)`
  display: none;
  color: #808080;
  @media screen and (max-width: 768px) {
    display: block;
    position: absolute;
    top: 0;
    right: 0;
    transform: translate(-100%, 75%);
    font-size: 1.8rem;
    cursor: pointer;
  }
`;

export const NavMenu = styled.div`
  display: flex;
  align-items: center;
  /* Second Nav */
  /* margin-right: 24px; */
  /* Third Nav */
  /* width: 100vw;
  white-space: nowrap; */
  @media screen and (max-width: 768px) {
    display: none;
  }
`;

// Sign in button UNUSED RN
export const NavBtn = styled.nav`
  display: flex;
  align-items: center;
  margin-right: 24px;
  font-family: Segoe UI;
  /* Third Nav */
  /* justify-content: flex-end;
  width: 100vw; */
  @media screen and (max-width: 768px) {
    display: none;
  }
`;

// Sign in button style UNUSED RN
export const NavBtnLink = styled(Link)`
  border-radius: 4px;
  background: #808080;
  padding: 10px 22px;
  color: #000000;
  outline: none;
  border: none;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  text-decoration: none;
  /* Second Nav */
  margin-left: 24px;
  &:hover {
    transition: all 0.2s ease-in-out;
    background: #fff;
    color: #808080;
  }
`;