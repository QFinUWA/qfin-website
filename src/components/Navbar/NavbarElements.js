import { FaBars } from 'react-icons/fa';
import { NavLink as Link } from 'react-router-dom';
import styled from 'styled-components';
  
// Style of the navbar
export const Nav = styled.nav`
  // position: sticky;
  position: relative;
  background: #24CC54;
  height: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0px calc((100vw - 1000px) / 2);
  z-index: 12;
  padding: 4px;
  @media screen and (max-width: 768px) {
    display: grid;
    grid-template-rows: 3rem auto;
    place-items: center;
    height: minmax(100%, max-content);
    // height: max-content;
  }
`;

// Colour of text on nav bar
export const NavLink = styled(Link)`
  color: #313131;
  font-weight: bold;
  display: flex;
  align-items: center;
  text-decoration: none;
  padding: 0 1rem;
  height: 100%;
  cursor: pointer;
  &.active {
    color: #000000;
  }
  @media screen and (max-width: 768px) {
    justify-content: center;
    width: 300px;
    color: rgba(255, 255, 255, 0.8);
  }
`;

export const NavQFin = styled(Link)`
  color: #313131;
  font-size:15pt;
  font-weight: bold;
  display: flex;
  align-items: center;
  text-decoration: none;
  padding: 0 1rem;
  height: 100%;
  cursor: pointer;
  &.active {
    color: #000000;
  }
`;

// Icon for the hamburger menu
export const Bars = styled(FaBars)`
  color: #808080;
  background: #24CC54;
  height: 2rem;
  position: absolute;
  top: 0.75rem;
  right: 1.5rem;
  font-size: 1.8rem;
  cursor: pointer;
`;

export const NavMenu = styled.div`
  display: flex;
  align-items: center;
  /* Second Nav */
  /* margin-right: 24px; */
  /* Third Nav */
  /* width: 100vw;
  white-space: nowrap; */
  @media screen and (max-width: 768px) {
    align-items: right;
    float: right;
    margin-top: 4px;
    background-color: rgba(97, 97, 102, 0.9);
    width: auto;
    display: grid;
    grid-auto-rows: 1fr;
    grid-template-columns: 1fr;
    place-items: center;
    grid-gap: 0.5rem;
    padding-bottom: 1rem;
  }
`;

// Sign in button UNUSED RN
export const NavBtn = styled.nav`
  display: flex;
  align-items: center;
  margin-right: 24px;
  font-family: Segoe UI;
  /* Third Nav */
  /* justify-content: flex-end;
  width: 100vw; */
  @media screen and (max-width: 768px) {
    display: none;
  }
`;

// Sign in button style UNUSED RN
export const NavBtnLink = styled(Link)`
  border-radius: 4px;
  background: #808080;
  padding: 10px 22px;
  color: #000000;
  outline: none;
  border: none;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  text-decoration: none;
  /* Second Nav */
  margin-left: 24px;
  &:hover {
    transition: all 0.2s ease-in-out;
    background: #fff;
    color: #808080;
  }
`;

export const BurgerNavMenu = styled.div`
  position: absolute;
  justify-items: center;
  float: right;
  background-color: rgba(97, 97, 102, 0.9);
  width: 100%;
  display: grid;
  grid-auto-rows: 1fr;
  grid-template-columns: 1fr;
  place-items: center;
  grid-gap: 0.5rem;
  padding: 1rem 0rem 1rem 0rem;
  z-index: 10
  `