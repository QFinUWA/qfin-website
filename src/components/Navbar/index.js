import React from 'react';
import {
  Nav,
  NavLink,
  Bars,
  NavMenu,
  NavQFin,
  //NavBtn,
  //NavBtnLink,
} from './NavbarElements';
  
const Navbar = () => {
  return (
    <>
      <Nav>
        <NavQFin className="QFin" to="/home" activestyle="true">
          QFin UWA
        </NavQFin>
        <Bars />
        <NavMenu>
          <NavLink to='/about' activestyle="true">
            About
          </NavLink>
          <NavLink to='/projects' activestyle="true">
            Projects
          </NavLink>
          <NavLink to='/events' activestyle="true">
            Events
          </NavLink>
          <NavLink to='/team' activestyle="true">
            Teams
          </NavLink>
          <NavLink to='/blogs' activestyle="true">
            Blogs
          </NavLink>
          <NavLink to='/sign-up' activestyle="true">
            Sign Up
          </NavLink>
          {/* Second Nav */}
          {/* <NavBtnLink to='/sign-in'>Sign In</NavBtnLink> */}
        </NavMenu>
      </Nav>
    </>
  );
};
  
export default Navbar;