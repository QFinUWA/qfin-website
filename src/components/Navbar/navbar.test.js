import React from 'react';
import { render } from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom'
import userEvent from '@testing-library/user-event';

import '@testing-library/jest-dom';

import Navbar from './index';

describe('<Navbar/>', () => {
    it('matches the snapshot', () => {
        const { container } = render(<Navbar />, { wrapper: MemoryRouter });
        expect(container).toMatchSnapshot();
    });

    it('renders with all menu items on desktop', () => {
        const { getByTestId } = render(<Navbar />, { wrapper: MemoryRouter });
        const navbar = getByTestId('nav');
        expect(navbar.childElementCount).toBe(2);
        expect(navbar.children[1].childElementCount).not.toBe(1);
    });
})