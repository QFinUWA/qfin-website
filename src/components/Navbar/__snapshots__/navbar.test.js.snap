// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> matches the snapshot 1`] = `
<div>
  <nav
    class="sc-bdvvtL gZblpu"
    data-testid="nav"
  >
    <a
      activestyle="true"
      class="sc-dkPtRN iNUeEW QFin"
      href="/home"
    >
      QFin UWA
    </a>
    <div
      class="sc-eCImPb dvDiWV"
    >
      <a
        activestyle="true"
        class="sc-gsDKAQ cqOWjq"
        href="/about"
      >
        About
      </a>
      <a
        activestyle="true"
        class="sc-gsDKAQ cqOWjq"
        href="/projects"
      >
        Projects
      </a>
      <a
        activestyle="true"
        class="sc-gsDKAQ cqOWjq"
        href="/events"
      >
        Events
      </a>
      <a
        activestyle="true"
        class="sc-gsDKAQ cqOWjq"
        href="/team"
      >
        Team
      </a>
      <a
        activestyle="true"
        class="sc-gsDKAQ cqOWjq"
        href="/blogs"
      >
        Blogs
      </a>
      <a
        activestyle="true"
        class="sc-gsDKAQ cqOWjq"
        href="/sign-up"
      >
        Sign Up
      </a>
    </div>
  </nav>
</div>
`;
